### ISSUE TITLE
Fix Request: false positive in UseClearKeyGeneratorName

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#446]
```
import org.springframework.cache.interceptor.KeyGenerator;

public class CustomerCacheKeyGenerator implements KeyGenerator { // <-- false positive
   public Object generate(Object target, Method method, Object... params) {
        // build key and return it
   }
}
```
The class name is not checked as documented for the rule.
